### DISCUSSION TITLE
Any way to limit GitHub App installation to only our orgs?

### DISCUSSION THREAD

**Discussion by dtamez** [community/community#24556] [Enterprise]
<p>Our enterprise has over 40 orgs in it.  The aim of this particular app is to generate some reports on all of our repos. If I make the app public then anyone (not belonging to our orgs) could potentially install the app and that would skew the data we’re trying to gather.  If I make the app private then wouldn’t I have to create 40+ apps so that each org could then install it? I’m hoping there’s some middle ground or better approach.  </p>

**Comment by jeffwilcox** [comment, marked as answer]
<p>It’s on my wish list and I’ve shared that feedback with the ecosystem team.</p>
<p>Today, we have a 2-step process for apps at Microsoft related to repo inventory: anyone can install the app, but it’s not until it’s “adopted” by our system, where we mark the installation ID as confirmed Microsoft, that we actually collect any data or integrate it into systems.</p>
<p>It’s not ideal, but it then lets us run an occassional job to cleanup anything that seems out of place.</p>

**Reply by Th3S4mur41** [reply]
While this solution would work in our case too, it wouldn't prevent the app from calling the webhook eating up our quota even if the app would be doing nothing

**Comment by dtamez** [comment]
<p>Thanks!  We’ve pretty much come to the same conclusion.  So for now that’s the best way of dealing with this scenario.</p>

**Comment by hpsin** [comment]
Heya, this is supported via the `internal` visibility now for enterprise-owned github apps. https://github.blog/changelog/2024-10-22-enterprises-can-create-github-apps-for-use-within-the-enterprise/

**Reply by Th3S4mur41** [reply]
Too bad you have to be an enterprise owner to create them... 
Is there a security reason for this limitation?
